packages/lisk-transactions/src/utils/get_transaction_bytes.ts

```diff
@@ -12,53 +12,87 @@
  * Removal or modification of this copyright notice is prohibited.
  *
  */
-import bignum from 'browserify-bignum';
 import { MAX_TRANSACTION_AMOUNT } from '@liskhq/lisk-constants';
 import cryptography from '@liskhq/lisk-cryptography';
+import BigNum from 'browserify-bignum';
 import { BYTESIZES } from '../constants';
+import {
+	BaseTransaction,
+	DappAsset,
+	DelegateAsset,
+	InTransferAsset,
+	MultiSignatureAsset,
+	OutTransferAsset,
+	PartialTransaction,
+	SecondSignatureAsset,
+	TransactionAsset,
+	TransferAsset,
+	TransferTransaction,
+	VoteAsset,
+} from '../types/transactions';
 
-export const isValidValue = value => ![undefined, false, NaN].includes(value);
+// tslint:disable-next-line no-any
+export const isValidValue = (value: any): boolean =>
+	![undefined, false, NaN].includes(value);
 
-export const checkRequiredFields = (requiredFields, data) => {
+export const checkRequiredFields = (
+	requiredFields: ReadonlyArray<string>,
+	// tslint:disable-next-line no-any
+	data: { readonly [key: string]: any },
+): boolean => {
 	const dataFields = Object.keys(data);
 	requiredFields.forEach(parameter => {
-		if (
-			!dataFields.includes(parameter.toString()) ||
-			!isValidValue(data[parameter])
-		) {
+		if (!dataFields.includes(parameter) || !isValidValue(data[parameter])) {
 			throw new Error(`${parameter} is a required parameter.`);
 		}
 	});
+
 	return true;
 };
 
-export const getAssetDataForTransferTransaction = ({ data }) =>
-	data ? Buffer.from(data, 'utf8') : Buffer.alloc(0);
+export const getAssetDataForTransferTransaction = (
+	asset: TransactionAsset,
```

Comment:
type definition for these functions is not very strict. what do you think?

Reply:
I tried, but with the current structure, I couldn't find a good way to handle this.
When they refactor, we can make it more strict?

Reply:
Okay